src/history/index.test.js

```diff
@@ -1,66 +1,316 @@
+        expect(path).to.equal(EXPECTED_FILE_PATH);
+      });
+
+      it('returns a boolean to know if it is the first record', async () => {
+        expect(isFirstRecord).to.be.false;
+      });
+
+      it('returns the id of the commit', async () => {
+        expect(commit.hash).to.includes(id);
+      });
+
+      it('properly saves the commit message', async () => {
+        expect(commit.message).to.equal(UPDATE_COMMIT_MESSAGE);
+      });
+
+      context('when the content has not changed', () => {
+        before(async () => {
+          const { id: snapshotId, path: snapshotPath } = await recordSnapshot(SERVICE_ID, TYPE, MODIFIED_FILE_CONTENT);
+          id = snapshotId;
+          path = snapshotPath;
+          const commits = await gitSnapshot().log();
+          [commit] = commits;
+        });
+
+        it('does not commit', () => {
+          expect(id).to.be.undefined;
+        });
+      });
     });
   });
 
   describe('#recordVersion', () => {
+    const FILE_CONTENT = '# ToS fixture data with UTF-8 çhãràčtęrs';
+    const EXPECTED_FILE_PATH = `${VERSIONS_PATH}/${SERVICE_ID}/${TYPE}.md`;
+    const FIRST_COMMIT_MESSAGE = `Start tracking ${SERVICE_ID} ${TYPE}`;
+    const SNAPSHOTS_ID = 'snapshot short sha';
     let id;
-    const snapshotId = 'commit-id';
+    let commit;
+    let path;
+    let isFirstRecord;
+
     before(async () => {
-      const { id: versionId } = await recordVersion(SERVICE_ID, TYPE, VERSION_FILE_CONTENT, snapshotId);
+      const { id: versionId, path: versionPath, isFirstRecord: isFirstVersionRecord } = await recordVersion(SERVICE_ID, TYPE, FILE_CONTENT, SNAPSHOTS_ID);
       id = versionId;
+      path = versionPath;
+      isFirstRecord = isFirstVersionRecord;
+      const commits = await gitVersion().log();
+      [commit] = commits;
+    });
+
+    after(resetGitRepository);
+
+    it('creates the file with the proper content', () => {
+      expect(fs.readFileSync(EXPECTED_FILE_PATH, { encoding: 'utf8' })).to.equal(FILE_CONTENT);
     });
 
-    after(() => {
-      fs.unlinkSync(EXPECTED_VERSION_FILE_PATH);
+    it('returns the file path', () => {
+      expect(path).to.equal(EXPECTED_FILE_PATH);
     });
 
-    it('creates a file for the given service', () => {
-      expect(fs.readFileSync(EXPECTED_VERSION_FILE_PATH, { encoding: 'utf8' })).to.equal(VERSION_FILE_CONTENT);
+    it('returns a boolean to know if it is the first record', async () => {
+      expect(isFirstRecord).to.be.true;
     });
 
-    it('commits the file for the given service', () => {
-      expect(id).to.exist;
-      expect(id).to.be.a('string');
+    it('returns the id of the commit', async () => {
+      expect(commit.hash).to.includes(id);
+    });
+
+    it('properly saves the commit message', async () => {
+      expect(commit.message).to.equal(FIRST_COMMIT_MESSAGE);
+    });
+
+    it('properly adds snapshot’s id in the commit body', async () => {
+      expect(commit.body).to.includes(SNAPSHOTS_ID);
```

Comment:
(repeated in several occurrences)

```suggestion
      expect(commit.body).to.include(SNAPSHOTS_ID);
```